src/compiler.ts

```diff
@@ -5947,6 +5968,13 @@ export class Compiler extends DiagnosticEmitter {
       elementExpression,
       contextualType != Type.void
     );
+
+    if (target instanceof TypedElement) {
+      let typedTarget = <TypedElement>target;
```

Comment:
How about this?
```suggestion
      let typedTarget = target;
```
Btw it will be great remove such explicit casts to check type narrowing in compiler's bootsrtap as well. @dcodeIO  wdyt?

Reply:
Hmm, technically, one could even remove the additional `typedTarget` local. In turn, however, this makes me wonder what would happen with Wasm GC, where on the IR level `target` would still be of type `Element` and every use of `target` after narrowing would need to cast, which introduces a hidden cost to the developer, in particular in untouched builds.

What'd need to be done here is that upon narrowing, what `typedTarget = <TypedElement>target` did explicitly before would need to be done implicitly by the compiler, introducing a new local of the now-narrowed type - but we don't have the machinery for this yet. Quite unfortunate :/

Reply:
I think Most MVP of GC proposal will have `ref.cast_static` and friends. See this PR: https://github.com/WebAssembly/binaryen/issues/4149

Reply:
I don't think this hidden cost should block the semantics of the language. In any case, Wasm GC is in a very unstable phase and we don't yet know what it will end up looking like. It's too early to worry about that imho.

Reply:
I agree @MaxGraey. For TypeScript developer, automatically narrow type after `instanceof` is more clear than adding a local variant. Also, for most of cases, a static_cast / dynamic_cast after `instanceof` is needed because the developer will do it only when they want to use some method / member in the inherit class. After GC feature, this feature maybe make this cast from dynamic_cast to static_cast